File: sdks/python/bolt_prediction_sdk/models/enhanced_training_request.py
"""
EnhancedTrainingRequest 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class EnhancedTrainingRequest(SDKBaseModel):
    """增强版模型训练请求"""

    model_type: str = Field(description="模型类型: bolt/flange")
    node_id: Optional[Any] = Field(description="节点ID，空则训练所有", default=None)
    force_retrain: Optional[bool] = Field(description="是否强制重新训练", default=False)
    data_source: Optional[str] = Field(description="数据来源: db/csv/manual", default='db')
    is_incremental: Optional[bool] = Field(description="是否增量训练", default=False)
    base_model_version: Optional[Any] = Field(description="增量训练的基础版本", default=None)
    freeze_layers: Optional[Any] = Field(description="冻结的层名称", default=None)
    training_config: Optional[Any] = Field(description="详细训练配置", default=None)

File: sdks/java/src/main/java/com/boltprediction/sdk/model/EnhancedTrainingRequest.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 增强版模型训练请求 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class EnhancedTrainingRequest {

    @JsonProperty("model_type")
    private String modelType;

    @JsonProperty("node_id")
    private Object nodeId;

    @JsonProperty("force_retrain")
    private Boolean forceRetrain;

    @JsonProperty("data_source")
    private String dataSource;

    @JsonProperty("is_incremental")
    private Boolean isIncremental;

    @JsonProperty("base_model_version")
    private Object baseModelVersion;

    @JsonProperty("freeze_layers")
    private Object freezeLayers;

    @JsonProperty("training_config")
    private Object trainingConfig;

    public String getModelType() {
        return modelType;
    }

    public void setModelType(String modelType) {
        this.modelType = modelType;
    }

    public Object getNodeId() {
        return nodeId;
    }

    public void setNodeId(Object nodeId) {
        this.nodeId = nodeId;
    }

    public Boolean getForceRetrain() {
        return forceRetrain;
    }

    public void setForceRetrain(Boolean forceRetrain) {
        this.forceRetrain = forceRetrain;
    }

    public String getDataSource() {
        return dataSource;
    }

    public void setDataSource(String dataSource) {
        this.dataSource = dataSource;
    }

    public Boolean getIsIncremental() {
        return isIncremental;
    }

    public void setIsIncremental(Boolean isIncremental) {
        this.isIncremental = isIncremental;
    }

    public Object getBaseModelVersion() {
        return baseModelVersion;
    }

    public void setBaseModelVersion(Object baseModelVersion) {
        this.baseModelVersion = baseModelVersion;
    }

    public Object getFreezeLayers() {
        return freezeLayers;
    }

    public void setFreezeLayers(Object freezeLayers) {
        this.freezeLayers = freezeLayers;
    }

    public Object getTrainingConfig() {
        return trainingConfig;
    }

    public void setTrainingConfig(Object trainingConfig) {
        this.trainingConfig = trainingConfig;
    }

}